apis/r/src/rinterface.cpp

```diff
@@ -434,6 +434,114 @@ bool c_allows_dups(const std::string& uri, Rcpp::XPtr<somactx_wrap_t> ctxxp) {
     return sch->allows_dups();
 }
 
+// Taken from tiledb-r
+// https://github.com/TileDB-Inc/TileDB-R/blob/525bdfc0f34aadb74a312a5d8428bd07819a8f83/src/libtiledb.cpp#L323C1-L367C2
+const char *_tiledb_filter_to_string(tiledb_filter_type_t filter) {
+    switch (filter) {
+    case TILEDB_FILTER_NONE:
+        return "NONE";
+    case TILEDB_FILTER_GZIP:
+        return "GZIP";
+    case TILEDB_FILTER_ZSTD:
+        return "ZSTD";
+    case TILEDB_FILTER_LZ4:
+        return "LZ4";
+    case TILEDB_FILTER_RLE:
+        return "RLE";
+    case TILEDB_FILTER_BZIP2:
+        return "BZIP2";
+    case TILEDB_FILTER_DOUBLE_DELTA:
+        return "DOUBLE_DELTA";
+    case TILEDB_FILTER_BIT_WIDTH_REDUCTION:
+        return "BIT_WIDTH_REDUCTION";
+    case TILEDB_FILTER_BITSHUFFLE:
+        return "BITSHUFFLE";
+    case TILEDB_FILTER_BYTESHUFFLE:
+        return "BYTESHUFFLE";
+    case TILEDB_FILTER_POSITIVE_DELTA:
+        return "POSITIVE_DELTA";
+    case TILEDB_FILTER_CHECKSUM_MD5:
+        return "CHECKSUM_MD5";
+    case TILEDB_FILTER_CHECKSUM_SHA256:
+        return "CHECKSUM_SHA256";
+    case TILEDB_FILTER_DICTIONARY:
+        return "DICTIONARY_ENCODING";
+    case TILEDB_FILTER_SCALE_FLOAT:
+        return "SCALE_FLOAT";
+    case TILEDB_FILTER_XOR:
+        return "FILTER_XOR";
+    default: {
+        Rcpp::stop("unknown tiledb_filter_t (%d)", filter);
+    }
+    }
+}
+
+// internal helper function
+// filter options taken from tiledb-r
+// https://github.com/TileDB-Inc/TileDB-R/blob/525bdfc0f34aadb74a312a5d8428bd07819a8f83/src/libtiledb.cpp#L369-L388
+Rcpp::List _get_filter_opts(Rcpp::XPtr<tiledb::Filter> filter) {
+    auto filter_type = _tiledb_filter_to_string(filter->filter_type());
+
+    // catches are intentionally blank to leave the values as NA
+    int compression = R_NaInt, bytewidth = R_NaInt, factor = R_NaInt, offset = R_NaInt;
+    uint32_t bit_width = R_NaInt, positive_delta = R_NaInt;
+    try {filter->get_option(TILEDB_COMPRESSION_LEVEL, &compression);} catch (...) {};
```

Comment:
This is a catch-anything-at-all ... which can hide other errors we might not want to ignore ... @nguyenv may have advice here ... if not, we can dig a bit into the core API & see if there's a has-option we can call ...

Reply:
Looks like `Filter::to_str(filter)`
https://github.com/single-cell-data/TileDB-SOMA/blob/1.16.1/libtiledbsoma/test/unit_soma_dataframe.cc#L343

Reply:
@johnkerl I'm not sure `Filter::to_str()` would work here; I'm trying to get filter options, not their name. libtiledb throws an error if you try to get a non-existent option, hence the try-catch. I can't figure out what class the errors are (I'm not sure they are), and don't know how to parse an error message in C++, so this is why I have the catch-all

Reply:
ok let's chat a bit ... we do deconstruct core error messages quite a bit in libtiledbsoma so there is copyable precedent